TrafficSimulator/Assets/TrafficLights/TrafficLightController.cs

```diff
@@ -0,0 +1,121 @@
+using UnityEngine;
+
+enum Mode{FIRST, SECOND};
+
+public class TrafficLightController : MonoBehaviour
+{
+    [SerializeField] private TrafficLight[] _trafficLightsPair1; // Starts green
+    [SerializeField] private TrafficLight[] _trafficLightsPair2; // Starts red
+
+    private int _trafficLightsCount;
+
+    public float Delay = 10f;
+    float lastSwitchTime = 0f;
+
+    private Mode _currentMode = Mode.FIRST;
+
+    void Start()
+    {
+        _trafficLightsCount = _trafficLightsPair1.Length + _trafficLightsPair2.Length;
+    }
+
+    void Update()
+    {
+        if (Time.time - lastSwitchTime > Delay)
+        {
+            Controller(_trafficLightsPair1, _trafficLightsPair2);
+            lastSwitchTime = Time.time;
+        }
+    }
+
+    private void Controller(TrafficLight[] _trafficLightsPair1, TrafficLight[] _trafficLightsPair2)
```

Comment:
These are not private variables and so should be named `trafficLightPair1` and 2